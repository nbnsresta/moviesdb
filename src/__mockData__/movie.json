{
  "id": 680,
  "video": false,
  "vote_count": 17549,
  "vote_average": 8.5,
  "title": "Pulp Fiction",
  "release_date": "1994-09-10",
  "original_language": "en",
  "original_title": "Pulp Fiction",
  "genre_ids": [
    80,
    53
  ],
  "backdrop_path": "/suaEOtk1N1sgg2MTM7oZd2cfVp3.jpg",
  "adult": false,
  "overview": "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
  "poster_path": "/d5iIlFn5s0ImszYzBPb8JPIfbXD.jpg",
  "popularity": 33.662,
  "media_type": "movie"
}