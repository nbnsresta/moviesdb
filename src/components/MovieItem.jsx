import React from 'react';
import style from './movieitem.module.css'

const imageBaseUrl = process.env.REACT_APP_MOVIE_POSTER_BASE_PATH

export default ({movie}) => {
  return (
    <div className={style.movieItem} key={movie.id}>
      <img src={`${imageBaseUrl}${movie['poster_path']}`} alt={movie.title}/>
      <span className={style.movieTitle}>{movie.title}</span>
    </div>
  )
}