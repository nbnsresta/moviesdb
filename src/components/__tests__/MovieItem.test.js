import React from "react";
import { mount } from "enzyme";
import MovieItem from "../MovieItem";
import movie from "../../__mockData__/movie.json";

describe("Render | MovieItem", () => {
  it("Component renders properly", () => {
    const wrapper = mount(<MovieItem movie={movie} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
