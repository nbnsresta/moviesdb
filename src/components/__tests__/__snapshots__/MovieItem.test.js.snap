// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render | MovieItem Component renders properly 1`] = `"<div class=\\"movieItem\\"><img src=\\"http://image.tmdb.org/t/p/w185/d5iIlFn5s0ImszYzBPb8JPIfbXD.jpg\\" alt=\\"Pulp Fiction\\"><span class=\\"movieTitle\\">Pulp Fiction</span></div>"`;
